File: native/Travel Companion/Model/Firestore/FirestoreConstants.swift
//
//  FirestoreConstants.swift
//  Travel Companion
//
//  Created by Stefan Jaindl on 14.08.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//

import Foundation

class FirestoreConstants {
    struct Collections {
        static let users = "users"
        static let places = "places"
        static let plans = "plans"
        
        static let fligths = "fligths"
        static let publicTransport = "publicTransport"
        static let hotels = "hotels"
        static let restaurants = "restaurants"
        static let attractions = "attractions"
        
        static let photos = "photos"
    }
    
    struct Ids {
        struct Place {
            static let placeId = "placeId"
            static let latitude = "latitude"
            static let longitude = "longitude"
            static let name = "name"
        }
        
        struct Plan {
            static let name = "name"
            static let pinName = "pinName"
            static let startDate = "startDate"
            static let endDate = "endDate"
            static let imageReference = "image"
            
            static let path = "path"
        }
        
        struct User {
            static let userId = "uid"
            static let email = "email"
            static let displayName = "displayName"
            static let providerId = "providerID"
            static let photoUrl = "photoURL"
            static let phoneNumber = "phoneNumber"
        }
    }    
}

File: shared/src/commonMain/kotlin/com.sjaindl.travelcompanion/persistence/FirestoreConstants.kt
package com.sjaindl.travelcompanion.persistence

object FirestoreConstants {

    object Collections {
        internal val users = "users"
        internal val places = "places"
        internal val plans = "plans"
        internal val fligths = "fligths"
        internal val publicTransport = "publicTransport"
        internal val hotels = "hotels"
        internal val restaurants = "restaurants"
        internal val attractions = "attractions"
        internal val photos = "photos"
    }

    object Ids {
        object Place {
            internal val placeId = "placeId"
            internal val latitude = "latitude"
            internal val longitude = "longitude"
            internal val name = "name"
        }

        object Plan {
            internal val name = "name"
            internal val pinName = "pinName"
            internal val startDate = "startDate"
            internal val endDate = "endDate"
            internal val imageReference = "image"
            internal val path = "path"
        }

        object User {
            internal val userId = "uid"
            internal val email = "email"
            internal val displayName = "displayName"
            internal val providerId = "providerID"
            internal val photoUrl = "photoURL"
            internal val phoneNumber = "phoneNumber"
        }
    }
}
